### PR TITLE
Cast $raw to string to suppress ctype_digit deprecation

### LINKED ISSUE #59: PHP 8.1: ctype_digit(): Argument of type int will be interpreted as string in the future
The check in ``if ($raw === '' || ctype_digit($raw)) {`` throws a deprecation warning in PHP 8.1:

```
ctype_digit(): Argument of type int will be interpreted as string in the future
```

There are 2 places: AbstractEscaper:119 and HtmlEscaper:68

I think the best solution is to cast $raw to string in the beginning

### PR DESCRIPTION
**Pull request by arokettu**
Fixes #59 by casting $raw to string

### CODE REVIEW

File: src/Escaper/AbstractEscaper.php [hunk 1]
```diff
@@ -110,11 +110,13 @@ public function getEncoding()
      *
      * @param string $regex The regex to determine what characters to replace.
      *
-     * @return mixed The escaped value.
+     * @return string The escaped value.
```
Comment (harikt) [inline src/Escaper/AbstractEscaper.php:113]: I am ok with the change. We just need to make sure the raw value passed earlier were never array.

**Review by koriym — APPROVED** [review]

File: src/Escaper/AbstractEscaper.php (hunk 1 above)
Comment (arokettu) [reply src/Escaper/AbstractEscaper.php:113]: replace calls toUtf8(), toUtf8() calls convert(), convert() calls iconv() that does not accept non strings

Comment (arokettu) [reply src/Escaper/AbstractEscaper.php:113]: @harikt same with html escaper, it calls htmlspecialchars() that accepts only strings

Comment (harikt) [reply src/Escaper/AbstractEscaper.php:113]: Cool. Thank you for the inputs. I will look and merge. Thank you

**Comment by harikt** [comment]
Thank you @arokettu .

**Pull request merged**